functional/FunctionalTest.java

```diff
@@ -802,6 +872,9 @@ public static void main(String[] args) {
 
       presignedPostPolicy_test();
 
+      copyObject_test1();
+      copyObject_test2();
```

Comment:
you could have four or more functions here.

1. testing without any condition
2. testing with all condition
3. testing with specific condition
4. testing with combination of conditions like

```
Consideration 1 – If both of the x-amz-copy-source-if-match and x-amz-copy-source-if-unmodified-since headers are present in the request as follows:

x-amz-copy-source-if-match condition evaluates to true, and;

x-amz-copy-source-if-unmodified-since condition evaluates to false;

then, S3 returns 200 OK and copies the data.

 

Consideration 2 – If both of the x-amz-copy-source-if-none-match and x-amz-copy-source-if-modified-since headers are present in the request as follows:

x-amz-copy-source-if-none-match condition evaluates to false, and;

x-amz-copy-source-if-modified-since condition evaluates to true;

then, S3 returns 412 Precondition Failed response code. 
```